packages/flutter_tools/lib/src/base/common.dart

```diff
@@ -2,10 +2,25 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
 const int kDefaultObservatoryPort = 8100;
 const int kDefaultDiagnosticPort  = 8101;
 const int kDefaultDrivePort       = 8183;
 
+/// Return the absolute path of the user's home directory
+String get homeDirPath {
+  if (_homeDirPath == null) {
+    _homeDirPath = Platform.environment['HOME'] ?? Platform.environment['USERPROFILE'];
```

Comment:
Perhaps use a ternary expression against the os, or add this as a method in the `OperatingSystemUtils` class?